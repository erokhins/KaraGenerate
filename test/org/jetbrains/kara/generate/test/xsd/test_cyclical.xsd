<?xml version='1.0' encoding='utf-8' ?>
<xsd:schema xmlns='test' xmlns:xsd='http://www.w3.org/2001/XMLSchema' version="1.0" targetNamespace='test'>


    <xsd:element name="cyclicalElement1" type="cyclicalType"/>
    <xsd:complexType mixed="true" name="cyclicalType">
        <xsd:choice>
            <xsd:element ref="cyclicalElement1"/>
        </xsd:choice>
    </xsd:complexType>

    <xsd:group name="cyclicalGroup">
        <xsd:choice>
            <xsd:element ref="cyclicalElement2"/>
        </xsd:choice>
    </xsd:group>
    <xsd:element name="cyclicalElement2">
        <xsd:complexType mixed="true">
            <xsd:choice>
                <xsd:group ref="cyclicalGroup"/>
            </xsd:choice>
        </xsd:complexType>
    </xsd:element>
</xsd:schema>