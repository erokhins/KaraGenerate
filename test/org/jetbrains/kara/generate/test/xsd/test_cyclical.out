AttributeTypeDecl:
AttributeDecl:
AttributeGroupDecl:
ElementDecl:
   ElementDeclaration = (cyclicalElement1, false)
   ElementGroups = [cyclicalType]
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =

   ElementDeclaration = (cyclicalElement2, true)
   ElementGroups = [cyclicalGroup]
   NewAllowElements = []
   AttributeGroups = []
   NewAttributes =

ElementGroup:
   ElementDeclaration = (cyclicalGroup, false)
   ElementGroups = []
   NewAllowElements = [cyclicalElement2]
   AttributeGroups = []
   NewAttributes =

   ElementDeclaration = (cyclicalType, true)
   ElementGroups = []
   NewAllowElements = [cyclicalElement1]
   AttributeGroups = []
   NewAttributes =

